weed/command/scaffold/filer.toml

```diff
@@ -41,13 +41,13 @@ enabled = false
 dbFile = "./filer.db"                # sqlite db file
 
 [mysql]  # or memsql, tidb
-# CREATE TABLE IF NOT EXISTS filemeta (
-#   dirhash     BIGINT               COMMENT 'first 64 bits of MD5 hash value of directory field',
-#   name        VARCHAR(1000) BINARY COMMENT 'directory or file name',
-#   directory   TEXT BINARY          COMMENT 'full path to parent directory',
-#   meta        LONGBLOB,
-#   PRIMARY KEY (dirhash, name)
-# ) DEFAULT CHARSET=utf8;
+# CREATE TABLE IF NOT EXISTS `filemeta` (
+#   `dirhash`   BIGINT NOT NULL       COMMENT 'first 64 bits of MD5 hash value of directory field',
+#   `name`      VARCHAR(766) NOT NULL COMMENT 'directory or file name',
+#   `directory` TEXT NOT NULL         COMMENT 'full path to parent directory',
+#   `meta`      LONGBLOB NOT NULL,
+#   PRIMARY KEY (`dirhash`, `name`)
+# ) DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

Comment:
are these charset and collate changes backward compatible?

Reply:
Setting the charsets here only affects newly created tables. In that regard the changes are backward compatible. The only issue could arise if someone runs an "ancient" mysql version (prior to 5.5 which was released December 3, 2010) which doesn't have the utf8mb4 type. In that case the table creation would fail.
I haven't tested all different mysql protocol supporting servers (probably not really possible anyway) but if some of them support utf8 and not utf8mb4 it would also fail on them. My guess is however that this shouldn't be the case as utf8mb4 is around for ages. I already used `utf8mb4_bin` instead of `utf8mb4_0900_bin` to be as compatible as possible, even though sorting would be faster with `utf8mb4_0900_bin`